src/scripts.js

```diff
@@ -8,174 +8,84 @@ import Recipe from './recipe';
-searchInput.addEventListener('input', updateSearch);
-tagContainer.addEventListener('click', findRecipeBytag);
-
-function onStartup() {
-  populateCookBook();
-  populateUserData();
-}
+homeButton.addEventListener('click', homeHandler);
+cardArea.addEventListener('click', cardButtonConditionals);
+favButton.addEventListener('click', determineFavorites);
+searchInput.addEventListener('input', domUpdates.updateSearch);
+tagContainer.addEventListener('click', tagHandler);
 
-function populateCookBook() {
-  let ingredients, recipes;
-  Cookbook.getIngredients()
-    .then(data => ingredients = data)
-  Cookbook.getRecipes()
-    .then(data => recipes = data)
-    .then(() => new Cookbook(recipes, ingredients))
-    .then(data =>  domUpdates.populateCards(data.recipes.recipeData))
-}
 
-function populateUserData() {
+function onStartup() {
   let userId = (Math.floor(Math.random() * 49) + 1)
-  let currentUser, userPantry
-  User.getUserData(userId) 
-    .then(data => currentUser = data)
-    .then(() => greetUser(currentUser))
-    .then(() => userPantry = new Pantry(currentUser.pantry))
-    .then(() => console.log(userPantry))
-}
-
-function viewFavorites() {
-  if (cardArea.classList.contains('all')) {
-    cardArea.classList.remove('all')
-  }
-  if (!user.favoriteRecipes.length) {
-    favButton.innerHTML = 'You have no favorites!';
-    populateCards(cookbook.recipes);
-    return
-  } else {
-    favButton.innerHTML = 'Refresh Favorites'
-    cardArea.innerHTML = '';
-    user.favoriteRecipes.forEach(recipe => {
-      cardArea.insertAdjacentHTML('afterbegin', 
-        `
-      <div id='${recipe.id}'class='card'>
-      <header id='${recipe.id}' class='card-header'>
-      <div class='header-container'>
-        <label for='add-button' class='hidden'></label>
-        <button id='${recipe.id}' aria-label='add-button' class='add-button card-button'>
-          <img id='${recipe.id} favorite' class='add'
-          src='https://image.flaticon.com/icons/svg/32/32339.svg' alt='Add to
-          recipes to cook'>
-        </button>
-        <label for='favorite-button' class='hidden'>Click to favorite recipe
-        </label>
-        <button id='${recipe.id}' aria-label='favorite-button' class='favorite favorite${recipe.id} card-button'></button>
-      </div>
-      <span id='${recipe.id}' class='recipe-name'>${recipe.name}</span>
-      <img id='${recipe.id}' tabindex='0' class='card-picture'
-        src='${recipe.image}' alt='click to view recipe for ${recipe.name}'>
-      </header>
-  </div>
-      `)
+  let promise1 = Cookbook.getIngredients()
+  let promise2 = Cookbook.getRecipes()
+  let promise3 = User.getUserData(userId) 
+  Promise.all([promise1, promise2, promise3])
+    .then(values => {
+      cookbook = new Cookbook(values[1].recipeData, values[0].ingredientsData)
+      currentUser = new User(values[2])
+      domUpdates.greetUser(currentUser)
+      domUpdates.populateCards(cookbook.recipes, cardArea)
```

Comment:
All promises are now handled in Promise.all, as these all need to be resolved for the information to be correctly displayed on the DOM

Reply:
I have been reading more about Promise.all and I like the facto that the set time out is not required anymore and it also resolves the issue we had about not know when the promises would resolve. I think it was a perfect addition!